src/driver/hermes_cdev.c

```diff
@@ -97,6 +104,105 @@ static int hermes_close(struct inode *inode, struct file *filp)
 	return 0;
 }
 
+static int __cmd_comp(struct hermes_dev *hermes, int eng)
+{
+	union hermes_cmd_ctrl ctrl;
+	ctrl.ehcmdctrl = ioread8(&hermes->cmds_ctrl[eng].ehcmdctrl);
```

Comment:
Why is this a crazy union?

Can't it just be `return ioread8()`?

Reply:
The spec currently assigns each bit of the EHCMDCTRL register a different meaning https://github.com/Eideticom/eid-hermes/blob/master/specs/eid-hermes-interface.md#ehcmdctrl

Hence the bit field in the other comment... But I guess we have plenty of space in the BAR so that both EHCMDEXEC and EHCMDDONE can take a byte each. I'll submit a separate PR for that

Reply:
It can be a bit field just fine.... But then you should really use a & MASK or some such. I still don't understand why it's a union.